ogr/ogrsf_frmts/hana/ogrhanatablelayer.cpp

```diff
@@ -0,0 +1,1762 @@
+                statement.setString(
+                    paramIndex, featReader.GetFieldAsDoubleArray(fieldIndex));
+            else
+                statement.setDouble(
+                    paramIndex, featReader.GetFieldAsDouble(fieldIndex));
+            break;
+        case odbc::SQLDataTypes::Decimal:
+        case odbc::SQLDataTypes::Numeric:
+            if ((!feature->IsFieldSet(fieldIndex)
+                 || feature->IsFieldNull(fieldIndex))
+                && feature->GetFieldDefnRef(fieldIndex)->GetDefault()
+                       == nullptr)
+                statement.setDecimal(paramIndex, odbc::Decimal());
+            else
+                statement.setDouble(
+                    paramIndex, featReader.GetFieldAsDouble(fieldIndex));
+            break;
+        case odbc::SQLDataTypes::Char:
+        case odbc::SQLDataTypes::VarChar:
+        case odbc::SQLDataTypes::LongVarChar:
+            if (clmDesc.isArray)
+                statement.setString(
+                    paramIndex, featReader.GetFieldAsStringArray(fieldIndex));
+            else
+                statement.setString(
+                    paramIndex,
+                    featReader.GetFieldAsString(fieldIndex, clmDesc.length));
+            break;
+        case odbc::SQLDataTypes::WChar:
+        case odbc::SQLDataTypes::WVarChar:
+        case odbc::SQLDataTypes::WLongVarChar:
+            if (clmDesc.isArray)
+                statement.setString(
+                    paramIndex, featReader.GetFieldAsStringArray(fieldIndex));
+            else
+                statement.setString(
+                    paramIndex,
+                    featReader.GetFieldAsNString(fieldIndex, clmDesc.length));
+            break;
+        case odbc::SQLDataTypes::Binary:
+        case odbc::SQLDataTypes::VarBinary:
+        case odbc::SQLDataTypes::LongVarBinary: {
+            Binary bin = featReader.GetFieldAsBinary(fieldIndex);
+            statement.setBytes(paramIndex, bin.data, bin.size);
+        }
+        break;
+        case odbc::SQLDataTypes::DateTime:
+        case odbc::SQLDataTypes::TypeDate:
+            statement.setDate(
+                paramIndex, featReader.GetFieldAsDate(fieldIndex));
+            break;
+        case odbc::SQLDataTypes::Time:
+        case odbc::SQLDataTypes::TypeTime:
+            statement.setTime(
+                paramIndex, featReader.GetFieldAsTime(fieldIndex));
+            break;
+        case odbc::SQLDataTypes::Timestamp:
+        case odbc::SQLDataTypes::TypeTimestamp:
+            statement.setTimestamp(
+                paramIndex, featReader.GetFieldAsTimestamp(fieldIndex));
+            break;
+        }
+    }
+
+    for (std::size_t i = 0; i < geomColumns_.size(); ++i)
+    {
+        ++paramIndex;
+        Binary wkb{nullptr, 0};
+        OGRErr err = GetGeometryWkb(feature, static_cast<int>(i), wkb);
+        if (OGRERR_NONE != err)
+            return err;
+        statement.setBytes(paramIndex, wkb.data, wkb.size);
```

Comment:
is wkb.data CPLFree()'d after usage ?

Reply:
No, there is no need in doing this.
HANA layers has an internal member  std::vector<char> dataBuffer_, that is used for reading/writing of binary/string data. The wkb.data member just points to dataBuffer_.data().